### PR TITLE
feat: add expiration time buffer

### LINKED ISSUE #8: Add a minimum time that must pass between the measurement timestamp and the expiration date of the KPI token
Right now there's a check that validates that the specified measurement timestamp is not after the attached KPI token's expiration date. This is not enough, as if the timestamp is set one second before the expiration date the campaign would still de facto be completely useless as it would almost certainly send back all of the rewards to the campaign creator. Best case, this is just extremely annoying for KPI token holders. Worst case, users could lose funds if the KPI tokens are sold by a malicious entity.

Fix this!

### PR DESCRIPTION
**Pull request by luzzif**

### CODE REVIEW

**Comment by luzzif** [comment]
Closes https://github.com/carrot-kpi/defillama-oracle-template/issues/8

File: packages/contracts/src/DefiLlamaOracle.sol [hunk 1]
```diff
@@ -34,8 +37,14 @@ contract DefiLlamaOracle is BaseOracle, ConstrainedOracle, ConstantAnswererTrust
     /// of an oracle to when the final measurement should be taken. If for example we have a
     /// `_minimumElapsedTime` with value 30, and oracle created at t0 must have a measurement
```
Comment (guerrap) [inline packages/contracts/src/DefiLlamaOracle.sol:38]: Noticing a small typo here `and oracle` should be `an oracle`. Also we need to add the documentation for the new `_expirationBufferTime` param.

File: packages/contracts/test/InitializeTest.sol [hunk 2]
```diff
@@ -202,6 +202,18 @@ contract InitializeTest is BaseTestSetup {
         );
     }
```
Comment (guerrap) [inline packages/contracts/test/InitializeTest.sol:204]: Should we add the tests cases for these reverts too?

```sol
if (_minimumElapsedTime == 0) revert InvalidMinimumElapsedTime();
if (_expirationBufferTime == 0) revert InvalidExpirationBufferTime();
```

**Review by guerrap — CHANGES_REQUESTED** [review]

File: packages/contracts/src/DefiLlamaOracle.sol (hunk 1 above)
Comment (luzzif) [reply packages/contracts/src/DefiLlamaOracle.sol:38]: Addressed in ecf5fa2ebf50e5a6823f36c7b643d4ff95b68109

File: packages/contracts/test/InitializeTest.sol (hunk 2 above)
Comment (luzzif) [reply packages/contracts/test/InitializeTest.sol:204]: Addressed in ca2ecb52ced3d2a33e844b2b546c9efb966dc707

**Review by guerrap — APPROVED** [review]

**Pull request merged**